wip/leetcode-2-add-two-numbers.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * https://leetcode.com/problems/add-two-numbers/
+ */
+
+/**
+ * Definition for singly-linked list.
+ * function ListNode(val) {
+ *     this.val = val;
+ *     this.next = null;
+ * }
+ */
+/**
+ * @param {ListNode} l1
+ * @param {ListNode} l2
+ * @return {ListNode}
+ */
+var addTwoNumbers = function(l1, l2) {
+    const resultHead = { val: (l1.val || 0) + (l2.val || 0) }
```

Comment:
It's kind of a common problem, I guess. Inside the loop, they create a new node and set as `next` to the previous one but what if the result list contains only one node, right?

Actually, what I'm going to describe next you've already mentioned in your video.

In order to not make it as a corner case, and make the whole algorithm concise, they usually create a "fake head" node and return its next in the end. In that case, if the loop runs only once they still get the right answer.

Reply:
Done. It works like a charm. I guess I haven't internalized this idea unless you told me it's a common practice 🤘